packages/amplify-java-runtime-provider/src/utils/build.ts

```diff
@@ -0,0 +1,44 @@
+import path from 'path';
+import fs from 'fs-extra';
+import childProcess from 'child_process';
+import { BuildRequest, BuildResult } from 'amplify-function-plugin-interface';
+
+// copied from the existing build-resources.js file in amplify-cli with changes for new interface
+export async function buildResource(request: BuildRequest): Promise<BuildResult>{
+  const resourceDir = path.join(request.srcRoot);
+
+  if (!request.lastBuildTimestamp || isValidProject(resourceDir)) {
+    installDependencies(resourceDir);
+    return Promise.resolve({ rebuilt: true });  }
```

Comment:
formatting. Also this function should return rebuilt: true iff it did a fresh install / compile. Otherwise return rebuilt: false